aspnetcore/performance/caching/memory.md

```diff
@@ -1,164 +1,111 @@
-  .SetAbsoluteExpiration(TimeSpan.FromHours(1))
-```
-
-By default, an instance of `MemoryCache` will automatically manage the items stored, removing entries when necessary in response to memory pressure in the app. You can influence the way cache entries are managed by setting their `CacheItemPriority` when adding the item to the cache. For instance, if you have an item you want to keep in the cache unless you explicitly remove it, you would use the `NeverRemove` priority option:
-
-```csharp
-// keep item in cache indefinitely unless explicitly removed
-new MemoryCacheEntryOptions()
-  .SetPriority(CacheItemPriority.NeverRemove))
-```
+Request the `IMemoryCache` instance in the constructor:
 
-When you do want to explicitly remove an item from the cache, you can do so easily using the `Remove` method:
+```c#
+  public class HomeController : Controller
+  {
 
-```csharp
-cache.Remove(cacheKey);
+      public HomeController(IMemoryCache memoryCache)
+      {
+          _memoryCache = memoryCache;
+      }
 ```
 
-## Cache Dependencies and Callbacks
+`IMemoryCache` requires NuGet package "Microsoft.Extensions.Caching.Memory".
 
-You can configure cache entries to depend on other cache entries, the file system, or programmatic tokens, evicting the entry in response to changes. You can register a callback, which will run when a cache item is evicted.
+The following code uses [TryGetValue](https://docs.microsoft.com/en-us/aspnet/core/api/microsoft.extensions.caching.memory.imemorycache) to cache and fetch the current time:
 
-[!code-csharp[Main](memory/sample/test/CachingSample.Tests/MemoryCacheTests.cs?highlight=6-11,18&range=22-41)]
+[!code-csharp[Main](memory/sample/WebCache/Controllers/HomeController.cs?name=snippet1)]
 
-The callback is run on a different thread from the code that removes the item from the cache.
+The current time and the cached time is displayed:
 
->[!WARNING]
-> If the callback is used to repopulate the cache it is possible other requests for the cache will take place (and find it empty) before the callback completes, possibly resulting in several threads repopulating the cached value.
+[!code-html[Main](memory/sample/webcache/views/home/index.cshtml)]
 
-Possible `eviction reasons` are:
+The `DateTime.Now` value will stay in the cache while there are requests within the timeout period (and not enough memory pressure to evict the cached entry). The image below shows the cached value is older than the current value:
 
-**None**
+![Index view with two different times displayed](memory/_static/time.png)
 
-No reason known.
+The following code uses [GetOrCreate](https://docs.microsoft.com/en-us/aspnet/core/api/microsoft.extensions.caching.memory.cacheextensions) and [GetOrCreateAsync](https://docs.microsoft.com/en-us/aspnet/core/api/microsoft.extensions.caching.memory.cacheextensions) to cache data. 
 
-**Removed**
+[!code-csharp[Main](memory/sample/WebCache/Controllers/HomeController.cs?name=snippet2&highlight=3,13-14)]
```

Comment:
I'm having trouble building this branch at the moment but from the code it looks likes like you should be highlighting lines 3, 14-15?